lib/ipmi_delloem.c

```diff
@@ -4178,3 +4194,111 @@ ipmi_delloem_setled_main(struct ipmi_intf * intf, int argc, char ** argv)
 	/* Set drive LEDs */
 	return ipmi_setled_state (intf, bayId, slotId, mask);
 }
+/*
+ * Function Name: oem_dell_idracvalidator_command
+ * 
+ * Description:	This function returns the iDRAC6 type
+ * Input:			intf - ipmi interface
+ *
+ * Output:
+ * 
+ * Return:		0 -Success
+ *             -1 - Failure/Error
+*/
+static
+int
+oem_dell_idracvalidator_command(struct ipmi_intf *intf)
+{
+	uint8_t msg_data[MSG_LEN_4] = {0};
+	struct ipmi_rs *rsp = NULL;
+
+	msg_data[IDX_0] = 0; // get cmd
+	msg_data[IDX_1] = IDRAC_VALIDATOR_PARAM; // parameter
+	msg_data[IDX_2] = 2; // block selector
+	msg_data[IDX_3] = 0; // set selector
+
+	if (oem_dell_frame_send_request(intf, IPMI_NETFN_APP, 
+		IPMI_GET_SYS_INFO, msg_data, MSG_LEN_4, &rsp) != 0) {
+		return -1;
+	}
+	switch (rsp->data[IDX_10]) {
+	case IMC_IDRAC_11G_MONOLITHIC:
+	case IMC_IDRAC_11G_MODULAR:
+	case IMC_MASER_LITE_BMC:
+	case IMC_MASER_LITE_NU:
+		iDRAC_FLAG = IDRAC_11G;
+		break;
+	case IMC_IDRAC_12G_MONOLITHIC:
+	case IMC_IDRAC_12G_MODULAR:
+		iDRAC_FLAG = IDRAC_12G;
+		break;
+	case IMC_IDRAC_13G_MONOLITHIC:
+	case IMC_IDRAC_13G_MODULAR:
+	case IMC_IDRAC_13G_DCS:
+		iDRAC_FLAG = IDRAC_13G;
+		break;
+	case IMC_IDRAC_14G_MONOLITHIC:
+	case IMC_IDRAC_14G_MODULAR:
+	case IMC_IDRAC_14G_DCS:
+		iDRAC_FLAG = IDRAC_14G;
+		break;
+	case IMC_IDRAC_15G_MONOLITHIC:
+	case IMC_IDRAC_15G_MODULAR:
+	case IMC_IDRAC_15G_DCS:
+		iDRAC_FLAG = IDRAC_15G;
+		break;		
+	default:
+		iDRAC_FLAG = 0;
+		break;
+	}
+	IMC_Type = rsp->data[IDX_10];
+	return 0;
+}
+/*
+ * Function Name:       ipmi_delloem_main
+ *
+ * Description:         This function processes the delloem command
+ * Input:               intf    - ipmi interface
+ *                      argc    - no of arguments
+ *                      argv    - argument string array
+ * Output:
+ *
+ * Return:              return code     0 - success
+ *                                      -1 - failure
+ */
+int
+ipmi_delloem_main(struct ipmi_intf * intf, int argc, char ** argv)
+{
+	int rc = 0;
+	current_arg = 0;
+	if (!argc || !strcmp(argv[0], "help")) {
+		usage();
+		return 0;
+	}
+	oem_dell_idracvalidator_command(intf);
+	switch (str2val(argv[0], oem_dell_commands)) {
```

Comment:
What exactly was the benefit of replacing string comparisons in the original code with this call to `str2val` and introduction of `oem_dell_commands` array that isn't used anywhere except this `switch`?

Reply:
Planning to refactor the complete delloem commands code, for better readability and modularity. 
This is the first patch to achieve that goal. Hence introduced oem_dell_commands.